Letterbook.Core/IAccountService.cs

```diff
@@ -1,11 +1,13 @@
 ﻿using Letterbook.Core.Models;
+using Letterbook.Core.Models.WebFinger;
 
 namespace Letterbook.Core;
 
 public interface IAccountService
 {
     public Account? RegisterAccount(string email, string handle);
     public Account? LookupAccount(string id);
+    public WebFingerJsonResourceDescriptor LookupAccount(WebFingerQueryTarget queryTarget);
```

Comment:
The idea here is to honour the domain, which is essentially the webfinger RFC.

https://datatracker.ietf.org/doc/html/rfc7033#section-4

So that means I am matching the concepts like "The JSON Resource Descriptor" and "query target" directly from there.

For me this means anyone can look at the code and understand it based on what they know already about the protocol.

"The JSON Resource Descriptor" seems different to `Profile` as well (?)

Reply:
Possibly this behaviour doesn't belong on `IAccountService` at all?

Reply:
Right. The information you need to build the webfinger json resources will come from Profiles. The adapter will need to map from the Profile domain object to webfinger format. We're not going to store webfinger json objects. We'll just build them on demand from the project's actual domain objects.

Reply:
Is [this more like what you're after](https://github.com/ben-biddington/Letterbook/blob/main/Letterbook.Api.Tests/Resources/WebFinger/WebFingerTests.cs#L63)?

```
[Fact]
    public async Task ItReturnsAJsonResourceDescriptor()
    {
        var profile = Profile.CreatePerson(new Uri("https://uss-voyager.example"), "coffee_nebula");

        _web.FakeAccountService.AlwaysReturn(profile);

        var reply = await _client.GetAsync("/.well-known/webfinger?resource=acct:coffee_nebula@uss-voyager.example");

        var body = await reply.Content.ReadAsStringAsync();

        body.MustMatchJson(@"
            {
              ""subject"": ""acct:coffee_nebula@uss-voyager.example"",
            }");
    }
```